### PR TITLE
feat(CX2587): add no insights message (artwork added from my collection tab)

### PR DESCRIPTION
**Pull request by dariakoko**
This PR resolves [CX-2587]

### Description

When the artwork is added from MyCollection tab and has no insights (TODO: check for insights when the backend is ready) - show the message on the top of MyCollection tab
FIgma: https://www.figma.com/file/l3plkt7gMSg4p1Or0E8SLC/My-Collection-2022?node-id=3639%3A78384

https://user-images.githubusercontent.com/36167539/169536924-1d5cf61d-9327-45de-8617-31c2981db601.mp4

### PR Checklist

- [x] I tested my changes on **iOS** / **Android**.
- [x] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.
- [ ] I added an [app state migration].
- [x] I hid my changes behind a [feature flag].

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[CX-2587]: https://artsyproduct.atlassian.net/browse/CX-2587?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Scenes/MyCollection/MyCollection.tsx [hunk 1]
```diff
@@ -94,65 +98,79 @@ const MyCollection: React.FC<{
...
-            )}
-          </Flex>
-        )
-      })
+                <Button
+                  data-test-id="add-artwork-button-non-zero-state"
+                  size="small"
+                  variant="fillDark"
+                  onPress={async () => {
+                    await AsyncStorage.setItem(TAB_WHERE_ADD_WORKS_BUTTON_WAS_PRESSED, MY_COLLECION)
```
Comment (olerichter00) [inline src/app/Scenes/MyCollection/MyCollection.tsx:131]: Can we do this with easy-peasy instead? I have the feeling this gets quite messy and we should avoid using `AsyncStorage` directly.

File: src/app/Scenes/MyCollection/MyCollection.tsx [hunk 2]
```diff
@@ -94,65 +98,79 @@ const MyCollection: React.FC<{
...
+                </Button>
+              </ArtworksFilterHeader>
+              {!!showNewWorksBanner && (
+                <Message
+                  variant="info"
+                  title="Your collection is growing"
+                  text="Based on your purchase history, we’ve added the following works."
+                  showCloseButton
+                  onClose={() =>
+                    AsyncStorage.setItem(HAS_SEEN_MY_COLLECTION_NEW_WORKS_BANNER, "true")
```
Comment (olerichter00) [inline src/app/Scenes/MyCollection/MyCollection.tsx:152]: Same here.

File: src/app/Scenes/MyCollection/MyCollection.tsx [hunk 3]
```diff
@@ -41,10 +41,13 @@ import { useTracking } from "react-tracking"
 import { useScreenDimensions } from "shared/hooks"
 import { ARTWORK_LIST_IMAGE_SIZE } from "./Components/MyCollectionArtworkListItem"
 import { MyCollectionArtworks } from "./MyCollectionArtworks"
+import { AddedArtworkHasNoInsightsMessage } from "./Screens/Insights/MyCollectionInsightsMessages"
 import { useLocalArtworkFilter } from "./utils/localArtworkSortAndFilter"
 import { addRandomMyCollectionArtwork } from "./utils/randomMyCollectionArtwork"
 
 export const HAS_SEEN_MY_COLLECTION_NEW_WORKS_BANNER = "HAS_SEEN_MY_COLLECTION_NEW_WORKS_BANNER"
+export const TAB_WHERE_ADD_WORKS_BUTTON_WAS_PRESSED = "TAB_WHERE_ADD_WORKS_BUTTON_WAS_PRESSED"
```
Comment (olerichter00) [inline src/app/Scenes/MyCollection/MyCollection.tsx:49]: Hmm, from looking at this constant I'm not able to tell what it does exactly 🤔

File: src/app/Scenes/MyCollection/Screens/Insights/MyCollectionInsightsMessages.tsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import { Message } from "palette"
+
+interface AddedArtworkHasNoInsightsMessageProps {
+  onClose: () => void
+}
+export const AddedArtworkHasNoInsightsMessage: React.FC<AddedArtworkHasNoInsightsMessageProps> = ({
```
Comment (olerichter00) [inline src/app/Scenes/MyCollection/Screens/Insights/MyCollectionInsightsMessages.tsx:6]: I wonder whether we should create a component that just wraps `Message` or if we should just use `Message` directly. What do you think?

File: src/app/store/config/visualClues.ts [hunk 5]
```diff
@@ -24,4 +24,7 @@ export const visualClues = defineVisualClues({
   CompleteCollectorProfileMessage: {
     description: "The message shown if the collector profile is incomplete",
   },
+  AddedArtworkHasNoInsightsMessage_MyCTab: {
```
Comment (olerichter00) [inline src/app/store/config/visualClues.ts:27]: What do you think about changing it to:

```suggestion
  AddArtworkWithoutInsightsMessage: {
```

?

**Review by olerichter00 — COMMENTED** [review]
This looks great! I'd just prefer to not use `AsyncStorage` directly and pretty much avoid this pattern wherever we can. @MounirDhahri what do you think?

File: src/app/Scenes/MyCollection/Screens/Insights/MyCollectionInsightsMessages.tsx (hunk 4 above)
Comment (dariakoko) [reply src/app/Scenes/MyCollection/Screens/Insights/MyCollectionInsightsMessages.tsx:6]: I had an idea of adding all Messages related to the insights to this file, to keep the main file clean

File: src/app/store/config/visualClues.ts (hunk 5 above)
Comment (dariakoko) [reply src/app/store/config/visualClues.ts:27]: `AddArtworkWithoutInsightsMessage_MyCTab` may be?
We will have three more clues, four in total
Two of them for Insights tab and two for MyCollection

Comment (olerichter00) [reply src/app/store/config/visualClues.ts:27]: Sounds good! Will they all be marked as seen independently or will they disappear together after the user has seen one of them?

File: src/app/Scenes/MyCollection/Screens/Insights/MyCollectionInsightsMessages.tsx (hunk 4 above)
Comment (olerichter00) [reply src/app/Scenes/MyCollection/Screens/Insights/MyCollectionInsightsMessages.tsx:6]: That's a very good idea 👍

File: src/app/Scenes/MyCollection/MyCollection.tsx (hunk 3 above)
Comment (dariakoko) [reply src/app/Scenes/MyCollection/MyCollection.tsx:49]: Yes, I know, I clearly have difficulties with names and descriptions 😄 
Renaming it as soon as getting inspo

This constant should tell from which tab the "Add Works" button was pressed
"activeTab" name is already used for other purposed

File: src/app/store/config/visualClues.ts (hunk 5 above)
Comment (dariakoko) [reply src/app/store/config/visualClues.ts:27]: Good question
I would say they should disappear independently.
I'll be working on all these messages next week and see how they appear in different scenarios

**Review by MrSltun — DISMISSED** [review]
LGTM! Great work Daria 👏 🌟

**Comment by dariakoko** [comment]
Thanks for your feedback!

I removed AsyncStorage and now check from which tab the button was pressed by passing enum Tab as a navigation param @olerichter00 I found it cleaner than easy-peasy

File: src/app/Scenes/MyCollection/MyCollection.tsx (hunk 3 above)
Comment (dariakoko) [reply src/app/Scenes/MyCollection/MyCollection.tsx:49]: @olerichter00 sounds good. We actually don't need this const any longer, I pass the tab name as a navigation prop now

**Review by olerichter00 — APPROVED** [review]
Looks good 👏

**Pull request merged**